### DISCUSSION TITLE
Combining two trained models.

### DISCUSSION THREAD

**Discussion by Kisjjw** [ultralytics/ultralytics#16077] [Q&A]
I would like to combine my trained YOLOv10.pt model with a trained dehazing model. What should I do? I want to first pass the image through the dehazing model to obtain a clear image, and then pass it on to the YOLO model. I want to merge these two models into one .pt model.

**Comment by glenn-jocher** [comment]
@Kisjjw to combine your models, you'll need to create a custom script that first processes the image with the dehazing model and then passes the output to the YOLOv10 model. Merging them into a single `.pt` file isn't straightforward, as they serve different purposes. Consider using a pipeline approach in your application code instead.

**Reply by Kisjjw** [reply]
Can I create a new Module, load two models into it, run them sequentially in the forward method, and then export the .pt model through Torch's API?

**Reply by glenn-jocher** [reply]
@Kisjjw yes, you can create a custom PyTorch module to load and run both models sequentially in the forward method. However, exporting them as a single `.pt` model might not be straightforward due to potential compatibility issues. Consider managing them separately in your application code.
